crates/re_types/src/components/half_extents2d_ext.rs

```diff
@@ -0,0 +1,71 @@
+use crate::datatypes::Vec2D;
+
+use super::{HalfExtents2D};
+
+impl HalfExtents2D {
+    #[inline]
+    pub const fn new(x: f32, y: f32) -> Self {
+        Self(Vec2D::new(x, y))
+    }
+
+    #[inline]
+    pub fn x(self) -> f32 {
+        self.0.x()
+    }
+
+    #[inline]
+    pub fn y(self) -> f32 {
+        self.0.y()
+    }
+
+    /// Width of a box using this half-extent.
+    #[inline]
+    pub fn width(self) -> f32 {
+        self.x() * 2.0
+    }
+
+    /// Height of a box using this half-extent.
+    #[inline]
+    pub fn height(self) -> f32 {
+        self.y() * 2.0
+    }
+
+    /// Double the half-extent to get the full extent.
```

Comment:
This sounds like an order 😬 

```suggestion
    /// The whole extent, i.e. twice the half extents.
```